libraries/chain/proposal_evaluator.cpp

```diff
@@ -21,22 +21,83 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#include <graphene/chain/database.hpp>
 #include <graphene/chain/proposal_evaluator.hpp>
 #include <graphene/chain/proposal_object.hpp>
-#include <graphene/chain/account_object.hpp>
-#include <graphene/chain/protocol/fee_schedule.hpp>
-#include <graphene/chain/exceptions.hpp>
-#include <graphene/chain/asset_evaluator.hpp>
 #include <graphene/chain/hardfork.hpp>
 
+namespace graphene { namespace chain {
 
-#include <fc/smart_ref_impl.hpp>
 
-namespace graphene { namespace chain {
+struct proposal_operation_hardfork_visitor
+{
+   typedef void result_type;
+   const fc::time_point_sec &block_time;
+
+   proposal_operation_hardfork_visitor(const fc::time_point_sec t) : block_time(t) {}
```

Comment:
To be consistent, please change the parameter to `(const fc::time_point_sec& t)`, or change the member variable definition to a value rather than a reference.